sr_gazebo_sim/src/gazebo_hardware_sim.cpp

```diff
@@ -0,0 +1,223 @@
+{
+  for (size_t i = 0; i < transmissions->size(); ++i)
+  {
+    if (j0_transmission_name == transmissions->at(i).type_)
+    {
+      if (transmissions->at(i).joints_.size() > 0)
+      {
+        const std::string joint_name = transmissions->at(i).joints_[0].name_;
+        std::string second_joint_name = joint_name;
+        second_joint_name[second_joint_name.size() - 1] = '2';
+        this->j2_j1_joints_[second_joint_name] = joint_name;
+      }
+    }
+  }
+
+  for (size_t i = transmissions->size() - 1; i != static_cast<size_t>(-1); --i)
+  {
+    if (transmissions->at(i).joints_.size() > 0)
+    {
+      const std::string joint_name = transmissions->at(i).joints_[0].name_;
+
+      if ((joint_name.size() > 0) && ('3' == joint_name[joint_name.size() - 1]))
+      {
+        std::string second_joint_name = joint_name;
+        second_joint_name[second_joint_name.size() - 1] = '2';
+
+        if (this->j2_j1_joints_.count(second_joint_name) > 0)
+        {
+          transmission_interface::TransmissionInfo new_transmission = transmissions->at(i);
+
+          fixJointName(&new_transmission.joints_, joint_name, second_joint_name);
+          fixJointName(&new_transmission.actuators_, joint_name, second_joint_name);
+          transmissions->push_back(new_transmission);
+        }
+      }
+    }
+  }
+}
+
+bool SrGazeboHWSim::isHandJoint(const std::vector<transmission_interface::TransmissionInfo> &transmissions,
+                                const std::string &joint_name) const
+{
+  // TODO(Andriy): Reimplement this function. It is using simple assumption that hand joint has one of
+  // two types of transmission from sr_mechanism_model package.
+  bool result = false;
+  for (size_t i = 0; i < transmissions.size(); ++i)
+  {
+    if ((transmissions[i].joints_.size() > 0)
+        && ((j0_transmission_name == transmissions[i].type_) || (simple_transmission_name == transmissions[i].type_)))
+    {
+      if (transmissions[i].joints_[0].name_ == joint_name)
+      {
+        result = true;
+      }
+    }
+  }
+
+  return result;
+}
+
+void SrGazeboHWSim::initializeFakeRobotState(const urdf::Model * const urdf_model,
+                                             const std::vector<transmission_interface::TransmissionInfo> &transmissions)
+{
+  this->fake_state_.robot_model_ = *urdf_model;
+
+  for (std::map<std::string, boost::shared_ptr<urdf::Joint> >::const_iterator it = urdf_model->joints_.begin();
+      it != urdf_model->joints_.end(); ++it)
+  {
+    if (this->isHandJoint(transmissions, it->first))
+    {
+      this->fake_state_.joint_states_[it->first].joint_ = it->second;
+      this->fake_state_.joint_states_[it->first].calibrated_ = true;
+    }
+  }
+
+  registerInterface(&this->fake_state_);
+  ROS_INFO_STREAM("Registered fake state");
+}
+
+void SrGazeboHWSim::registerSecondHardwareInterface(std::vector<transmission_interface::TransmissionInfo> transmissions)
+{
+  for (unsigned int j = 0; j < transmissions.size(); j++)
```

Comment:
For loops are size_t, unsigned int and unsigned. Can we stick to one or is it intentional?